src/main/java/fluke/Parser.java

```diff
@@ -0,0 +1,272 @@
+            LocalDate toDate = LocalDate.parse(to, DATE_TIME_FORMATTER);
+            return new Event(desc, isMarked, fromDate.toString(), toDate.toString());
+        }
+        throw new SaveFileParsingException();
+    }
+
+    /**
+     * Parses a Todo command.
+     * @param command Command given by the user
+     * @return the description of the todo task.
+     * @throws FlukeException if the description given is empty.
+     */
+    public static String parseTodoCommand(String command) throws FlukeException {
+        if (command.length() <= 5) {
+            // command is too short, description is invalid
+            throw new EmptyDescriptionException();
+        }
+        return command.substring(5);
+    }
+
+    /**
+     * Parses and validates a Deadline command
+     * @param command Command given by the user.
+     * @return a String array containing at 0: description, 1: (String) by date
+     * @throws FlukeException if the description given is invalid, or if the by date given is invalid.
+     */
+    public static String[] parseDeadlineCommand(String command) throws FlukeException {
+        if (command.length() <= 9) {
+            // command is too short, description is invalid
+            throw new EmptyDescriptionException();
+        }
+        String str = command.substring(9);
+        int byIndex = str.indexOf("/by");
+        if (byIndex < 0) {
+            throw new InvalidInputException();
+        }
+        String description = str.substring(0, byIndex - 1);
+        String by = str.substring(byIndex + 4);
+        return new String[]{description, by};
+    }
+
+    /**
+     * Parses and validates an Event command.
+     * @param command Command given by the user.
+     * @return a String array containing at 0: description, 1: (String) from date 2: to date.
+     * @throws FlukeException if any of the inputs are invalid.
+     */
+    public static String[] parseEventCommand(String command) throws FlukeException {
+        if (command.length() <= 6) {
+            // command is too short, description is invalid
+            throw new EmptyDescriptionException();
+        }
+        String str = command.substring(6);
+        int fromIndex = str.indexOf("/from");
+        int toIndex = str.indexOf("/to");
+        if (fromIndex < 0 || toIndex < 0) {
+            throw new InvalidInputException();
+        }
+        String description = str.substring(0, fromIndex - 1);
+        String from = str.substring(fromIndex + 6, toIndex - 1);
+        String to = str.substring(toIndex + 4);
+        return new String[]{description, from, to};
+    }
+
+    /**
+     * Parses a Delete command.
+     * @param command Command given by the user.
+     * @return index of the task in the task list.
+     * @throws FlukeException if the input is invalid.
+     */
+    public static int parseDeleteCommand(String command) throws FlukeException {
+        if (command.length() <= 7) {
```

Comment:
Is it easier to make use of split function in Java so you can count number of words instead of characters instead and make the comparison there? 

[Here is a link for more info about split strings](https://www.geeksforgeeks.org/split-string-java-examples/)

so for example 
```java 
String[] splitText = command.split(" "); // Split command string with spaces
if (splitText.length > 1) { 
   // ...
} 
```